### PR TITLE
Propose a different traditional ok implentation

### PR DESCRIPTION
**Pull request by blue-devil-93**
Partial rewrite of traditional ok based on yatu-ok-clearlooks hosted on gitlab.This approach  is  compatible to yaru on a code base level.

### CODE REVIEW

**Review by vkareh — CHANGES_REQUESTED** [review]
This removes the Makefile, making it impossible to use automake tools to install this theme. Please restore the files so that I can test using my regular scripts.

**Comment by blue-devil-93** [comment]
@vkareh it is now fixed so that it can be installed at least i think so.

File: desktop-themes/TraditionalOk/gtk-3.0/Makefile.am [hunk 1]
```diff
@@ -3,12 +3,7 @@ THEME_NAME=TraditionalOk
 themedir = $(datadir)/themes/$(THEME_NAME)/gtk-3.0
 
 theme_DATA = \
-	applications.css \
-	gtk.css \
-	gtk-widgets.css \
-	gtk-widgets-img.css \
-	mate-applications.css \
-	other-applications.css \
+	gtk.css \
```
Comment (vkareh) [inline desktop-themes/TraditionalOk/gtk-3.0/Makefile.am:6]: There's a syntax error in this line. Currently it terminates in a whitespace, but it shouldn't
```diff
diff --git a/desktop-themes/TraditionalOk/gtk-3.0/Makefile.am b/desktop-themes/TraditionalOk/gtk-3.0/Makefile.am
index b4c56e30..030a1852 100644
--- a/desktop-themes/TraditionalOk/gtk-3.0/Makefile.am
+++ b/desktop-themes/TraditionalOk/gtk-3.0/Makefile.am
@@ -3,7 +3,7 @@ THEME_NAME=TraditionalOk
 themedir = $(datadir)/themes/$(THEME_NAME)/gtk-3.0
 
 theme_DATA = \
-       gtk.css \ 
+       gtk.css \
        settings.ini
 
 %.ini: %.ini.in
```

**Review by vkareh — CHANGES_REQUESTED** [review]
The configure.ac file also needs to be updated:
```diff
diff --git a/configure.ac b/configure.ac
index ae4ec045..1f222f71 100644
--- a/configure.ac
+++ b/configure.ac
@@ -135,12 +135,12 @@ desktop-themes/Menta-border/metacity-1/Makefile
 desktop-themes/TraditionalGreen/Makefile
 desktop-themes/TraditionalGreen/gtk-2.0/Makefile
 desktop-themes/TraditionalGreen/gtk-3.0/Makefile
-desktop-themes/TraditionalGreen/gtk-3.0/img/Makefile
+desktop-themes/TraditionalGreen/gtk-3.0/assets/Makefile
 desktop-themes/TraditionalGreen/metacity-1/Makefile
 desktop-themes/TraditionalOk/Makefile
 desktop-themes/TraditionalOk/gtk-2.0/Makefile
 desktop-themes/TraditionalOk/gtk-3.0/Makefile
-desktop-themes/TraditionalOk/gtk-3.0/img/Makefile
+desktop-themes/TraditionalOk/gtk-3.0/assets/Makefile
 desktop-themes/TraditionalOk/metacity-1/Makefile
 desktop-themes/TraditionalOk/openbox-3/Makefile
 desktop-themes/TraditionalOk/xfwm4/Makefile
```

File: desktop-themes/TraditionalOk/gtk-3.0/assets/Makefile.am [hunk 2]
```diff
@@ -0,0 +1,133 @@
+THEME_NAME=TraditionalOk
+
+themedir = $(datadir)/themes/$(THEME_NAME)/gtk-3.0/assets
+
+theme_DATA = \
+   bullet-symbolic.svg \
```
Comment (vkareh) [inline desktop-themes/TraditionalOk/gtk-3.0/assets/Makefile.am:6]: This file outputs the following error on everyone of these lines:
```
desktop-themes/TraditionalOk/gtk-3.0/assets/Makefile.am:6: warning: whitespace following trailing backslash
```
Please remove trailing whitespace

**Comment by vkareh** [comment]
After removing traling whitespaces and fix Makefile path in configure.ac I was able to compile - however, now I'm getting the following error when trying to use the theme:
Screenshot at 2021-06-07 12-57-43

This can probably be solved by _not_ renaming the theme, as this now replaces the TraditionalOk theme entirely

**Comment by blue-devil-93** [comment]
Now it should work but I agree that naming it yaru ok and not replace traditional ok might be the cleaner option

File: desktop-themes/TraditionalOk/index.theme.in [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
-Comment=Port of Clearlooks-Phenix
+Name=Traditional-Ok
+Comment=Port of Clearlooks-Phenix and Traditional OK
 Encoding=UTF-8
 
 [X-GNOME-Metatheme]
-GtkTheme=TraditionalOk
-MetacityTheme=TraditionalOk
+GtkTheme=Traditional-Ok
+MetacityTheme=Traditional-Ok
```
Comment (vkareh) [inline desktop-themes/TraditionalOk/index.theme.in:9]: It will keep failing due to the mismatched name - revert this file and it should install correctly.

**Comment by vkareh** [comment]
After reverting the `index.theme.in` file I was able to install and use this correctly. Looks very good and hard to distinguish from the original TraditionalOk.

I noticed one regression when hovering over another theme, the highlight is black, rather than the original light blue, making the text hard to read:
Screenshot at 2021-06-07 13-12-17

**Comment by blue-devil-93** [comment]
@vkareh it should now be closer to gtk2. look. This bug you mentioned need some time to fix. But  could reproduce it. Thats good. Guess i can fix it this week.

**Comment by raveit65** [comment]
I prefer the old theme names.

File: desktop-themes/Makefile.am [hunk 4]
```diff
@@ -14,5 +14,6 @@ SUBDIRS = \
 	Menta \
 	Menta-border \
 	TraditionalGreen \
-	TraditionalOk
+        TraditionalOk \
```
Comment (vkareh) [inline desktop-themes/Makefile.am:17]: Indentation here is a bit off

**Comment by vkareh** [comment]
@blue-devil-93 I see you renamed your theme, I think that's a good idea if the plan is to keep them as separate. But since they are very similar (but not exactly), what is the plan? Is it to eventually replace the Traditional themes? Or is it something else? In reading #289 it seems like part of the idea is that since colors are soft-coded they can be replaced easily - any chance you can provide some variants?

**Comment by vkareh** [comment]
FWIW, the latest push works very well and having both themes (for now at least) allows me to compare by switching back and forth between them.

**Comment by blue-devil-93** [comment]
@VictorKareh
My plan was-/is to eventually replace Traditional ok. But that is for
later.Reason wass gtk4 support and closer to gtk2 . Softcoded colors
also.Do fix the one space to much later this day.
What needs to be looked at is if  the sassc  compiler command from
parse-sass.sh can be put in the makefile.

**Comment by blue-devil-93** [comment]
As long as its possible in gtk2 i can provide variants like green, purple,
orange, darker blue etc...  easily later.  Palette scss core color a and b.
Maybe the selected colors in colors.scss than executing the parse-sass.sh
and the new color variant is ready.

**Comment by raveit65** [comment]
Who like to maintain new themes?

**Comment by raveit65** [comment]
> But that is for later.Reason wass gtk4 support and closer to gtk2 . 

For me gtk3 style looks much better than old ugly gtk2 theming.

**Comment by blue-devil-93** [comment]
@raveit65
Your point of view is also true.Maintaining color shemes is work even if we
change only two color definitions. Not in short but in long term usage.

Regardless of the opinion that gtk3 looks better we need to keep it looking
similar to gtk2 and also 4. Even if gtk2 is nearly dead there are some
applications still using it one way or another. And when gtk4 apps start to
grow it needs to be compatible as well.

**Comment by raveit65** [comment]
Well, personal i never used Traditional themes so i don't mind what a maintainer want to do with them.
My babies are Menta and Submarine themes and i hope nobody has this weird idea to bump them back in the past ;)
But we need clearly a person who is responsible for new themes.
Sorry, i am not willing to maintain more themes.

**Comment by blue-devil-93** [comment]
I hope that the image of gtk4 screenshot will be shown here

**Comment by vkareh** [comment]
I think it looks fine so far. I'll use this theme today to see how it feels as compared with TradOK - my default theme is Trad-Green, so the color is a bit jarring, but I should be able to spot some differences here and there.

Who is the maintainer of this theme? I think it's @monsta if I recall correctly?

**Comment by raveit65** [comment]
> Who is the maintainer of this theme? I think it's......

Trad themes was maintained by zenwalker.

**Comment by vkareh** [comment]
@sc0w - maybe we can use your input here regarding the Traditional themes

@blue-devil-93 - any chance you can make a YaruGreen theme? I'm really trying the Ok/Blue one and struggling to adapt, but I'm very familiar with TraditionalGreen and would be able to spot regressions.

**Comment by vkareh** [comment]
@blue-devil-93 you're missing these changes for the YaruGreen theme to install:
```diff
~/d/m/mate-themes (⎇ pull/290 *) $ git diff
diff --git a/configure.ac b/configure.ac
index df6f9058..07cdcba4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -157,6 +157,7 @@ desktop-themes/YaruGreen/Makefile
 desktop-themes/YaruGreen/gtk-2.0/Makefile
 desktop-themes/YaruGreen/gtk-3.0/Makefile
 desktop-themes/YaruGreen/gtk-3.0/assets/Makefile
+desktop-themes/YaruGreen/metacity-1/Makefile
 marco-themes/Makefile
 icon-themes/Makefile
 icon-themes/ContrastHigh/Makefile
diff --git a/desktop-themes/YaruGreen/gtk-3.0/assets/Makefile.am b/desktop-themes/YaruGreen/gtk-3.0/assets/Makefile.am
index 3363e8eb..da701114 100644
--- a/desktop-themes/YaruGreen/gtk-3.0/assets/Makefile.am
+++ b/desktop-themes/YaruGreen/gtk-3.0/assets/Makefile.am
@@ -1,4 +1,4 @@
-THEME_NAME=YaruOk
+THEME_NAME=YaruGreen
 
 themedir = $(datadir)/themes/$(THEME_NAME)/gtk-3.0/assets
 
diff --git a/desktop-themes/YaruGreen/metacity-1/Makefile.am b/desktop-themes/YaruGreen/metacity-1/Makefile.am
index 0b6f9982..beb651e3 100644
--- a/desktop-themes/YaruGreen/metacity-1/Makefile.am
+++ b/desktop-themes/YaruGreen/metacity-1/Makefile.am
@@ -1,4 +1,4 @@
-THEME_NAME=TraditionalGreen
+THEME_NAME=YaruGreen
 
 themedir = $(datadir)/themes/$(THEME_NAME)/metacity-1
 
```

**Comment by blue-devil-93** [comment]
@vkareh now it should be ready to test.

**Comment by vkareh** [comment]
@blue-devil-93 - a bit of feedback since I've been using YaruGreen for a few hours. I don't have screenshots, but these are pretty general and fairly easy to identify:
- [ ] tabs are too tall
- [ ] left-side alignment on tabs is a few pixels off
- [ ] osd overlay too dark (when doing alt+tab, or volume up/down, for example)
- [ ] scroll bar color is too light
- [ ] scroll bar is missing grabber lines in the middle
- [ ] calculator buttons are too wide
- [ ] clock applet has a strange button outline
- [ ] popup menus seem too rounded for my taste, but that's probably subjective
- [ ] on Firefox, when typing an address in the bar, the drop-down text is blue instead of green
- [ ] panel applets are missing grabber (the little 3 dots that show up on unlocked applets)

I might update this list if I find other inconsistencies, but this is probably a good time to ask: if we replace the Traditional themes with your version, are you willing to maintain them, or do you plan on keeping these as separate themes? What is the ultimate goal here?

Also, @raveit65 - is @sc0w still involved in the MATE project?

**Comment by blue-devil-93** [comment]
@victorkareh
Easily doable:
Tabs height
Scrollbarcolors darker for green. Just need to check if its general
selected bg color or only scrollbar. Can be done.
scale handle.

Needs a litlle bit more time but is doable:
Panel applets grabber.
Tab aligment you mentioned is right optically in gtk4 so it schould be
doable in gtk3 as well.Osd overlay

Dont like to fix:
Scrollbar grabbing lines. On nvidia the thickness of the lines change when
i resize the window. Spent hours one trying out  and nothing worked well.
Read the code tried every option documented. Maybe on bigger screens it is
not so visible as on my computer. However on small screens it really is a
problem.

@vakareh
The thing is the optical differences are there and even if i fix these
things it won' t work like the old one in every case. At the moment i
believe its better to keep them seperate. Later it can be merged. Except
december-/january im able to provide bugfixes if possible and necessary.

**Comment by raveit65** [comment]
Also, @raveit65 - is @sc0w still involved in the MATE project?

No idea, i am so bored about people who left MATE in anger for no reason.
From my point of few it doesn't matter if we replace Trad themes or add simple 2 new themes.
In any case we need a person who is responsible for the replacement or the new added theme.

**Comment by raveit65** [comment]
@blue-devil-93 
> Dont like to fix:
Scrollbar grabbing lines. On nvidia the thickness of the lines change when
i resize the window. Spent hours one trying out  and nothing worked well.
Read the code tried every option documented. Maybe on bigger screens it is
not so visible as on my computer. However on small screens it really is a
problem.

Can you please post a screenshot for the issue?
Let me try to dig it down.

**Comment by blue-devil-93** [comment]
I did now readd the partially working grabbing lines  code and pixmap.

What i did to obtain these two images: resize the window with scrollbar and taking a screensot. For better understanding what i mean i did upscale them. Hope you se the color hift of the lines too. Did notice it on virtualbox and nvidia cards. Can`t say how it performs on radeon or intel. This bug is not possible to fix imho. Thats why o dropped the grabbing lines -/ handles on all my other themes. Even if  they are really nice. Miss them too sometimes but...

yaru-handle-1
yaru-handle-2

**Comment by raveit65** [comment]
Hmm, from my experience is there a border-image?
Handle border-images are buggy in gtk3 and never fixed, because Adwaita theme don't use them.

**Comment by blue-devil-93** [comment]
The color of the lines fade when resizing the window. If magnified i see that they get blurry.  Can only check if  using svg file solves it but do`thingk it does.

**Comment by raveit65** [comment]
Well, if it depends on desktop size i would say it is a graphic driver issue or mesa.

**Comment by thesquash** [comment]
> Scrollbar grabbing lines. On nvidia the thickness of the lines change when i resize the window. Spent hours one trying out and nothing worked well.

Ah, I ran into that trouble while creating a complete look-alike of GTK+ 2 Clearlooks.  The problem is that the grabber is set as the background for the scrollbar slider, the background image is centered, and the slider sometimes isn't an even number of pixels tall (usually when the window is sized unfortunately).  As such GTK aligns the background image on a "half-pixel" boundary, and the closest it can get to that is making the image cross pixel boundaries and thus fuzzing the image a little.  I don't have a solution to this, alas; wouldn't it be nice if GTK+ 3 still supported theming engines?

**Comment by blue-devil-93** [comment]
@thesquash perfectly explained. 

@vkareh
For the overlay i need a screenshot where to find it.
Some issues you mentioned should be fixed now. 
Tommorow i will apply the changes to yaru ok and also look at the popup menus.Then do create a diff against yaru and adwaita.
 It schould be possible to apply patches  from these  themes later if necessary .

**Comment by blue-devil-93** [comment]
[diff_from_yaru.txt](https://github.com/mate-desktop/mate-themes/files/6643421/diff_from_yaru.txt)
@vkareh here the diff-/patch against original yaru theme. it should be possible to apply most changes from yaru directly to yaru ok and green if necessary. 

Now most of the bugs you mentionend should be fixed., Please  correct  the list so i get a bit of overview what needs still to be done

@raveit65 or @thesquash  if you have the time please explain how i can find the overlays or what they are. Rest is done now. Calculator buttons are in Math Buttons but mate-calc doesn accept css codes.

**Comment by vkareh** [comment]
@blue-devil-93 
Thanks for fixing the various issues - I've updated the list in that previous checklist. Looking much better!

>  For the overlay i need a screenshot where to find it.

What do you mean? Just press Alt+Tab on your key board and you will see the overlay. Also works with pressing the volume buttons on your keyboard, if you have them.

**Comment by raveit65** [comment]
> Calculator buttons are in Math Buttons but mate-calc doesn accept css codes.

Did you inspect the mate-calc window with gtk3-inspector? It should show you the right css names for buttons and show you from which place in your css it is written.
Maybe a later setting override it?

**Comment by blue-devil-93** [comment]
@raveit65 i tried some possible combinations and used env
GTK_DEBUG=interactive but no real solution. Yes i could globally decrease
text-button width but i believe the yaru and adwaita devs had  an reason
for setting the space at the sides so big.

@victorkareh
I can' t find the overlay widgets. Since i just dont use them  even a small
screenshot  to find them would help a lot.
Tab gradient is easy, think about it next days.

**Comment by raveit65** [comment]
> but i believe the yaru and adwaita devs had an reason for setting the space at the sides so big.

Is an unknown reason an reason?  :)
I thought yaru was your theme?
We need an maintainer for new themes. I do not agree with adding new themes without a person who is responsible for it.

**Comment by blue-devil-93** [comment]
@raveit65
Yes im responsible for yarugreen and yaruok.
The button with needs to be fixed for mate-calc and not for any other
application. A button with a too small width is the inverse problem. That
we dont want too. Its even worse. Try it again later with gtk_debug
interactive to find the correct name if the bitton.

Overlay widgets still no screenshot bit i will wait until i find these
widgets or get a good screenshot what they are.Overlay  Notifications i
will check soon.

**Comment by raveit65** [comment]
> Yes im responsible for yarugreen and yaruok.

Thank you, i send you and invitation for the themer and artist group. After that you'll have write access to mate-themes.

**Comment by raveit65** [comment]
> Overlay widgets still no screenshot bit i will wait until i find these
widgets or get a good screenshot what they are.Overlay  Notifications i
will check soon

They can be seen in gtk3-widget-factory.

**Comment by blue-devil-93** [comment]
@victorkareh still no idea what overlays are but now they should be lighter
as requested.

**Comment by vkareh** [comment]
@blue-devil-93 I've attached a screenshot of two overlays happening at the same time: the volume and the alt+tab
Screenshot at 2021-06-16 09-46-07

**Comment by blue-devil-93** [comment]
@vkareh
Thx a lot. Now i can find them too.
This weekend i have the time to provide an bugfix. It  can be merged then
to the main themes repository i think.
Later we can either rename yaruok and yarugreen to traditionalok and
traditionalgreen or leave them as they are  as seperate themes.

**Comment by raveit65** [comment]
@blue-devil-93 @vkareh 
You're are talking about OSD
In css code `overlay` or `.osd` can be used. See our reference theme Menta.
https://github.com/mate-desktop/mate-themes/blob/master/desktop-themes/BlueMenta/gtk-3.0/gtk-widgets.css#L3900
For example
```
overlay.osd {
        	background-color: transparent;
        }
```

**Comment by vkareh** [comment]
Nice! I like how the OSD overlays look

**Comment by vkareh** [comment]
Only a few items left:
- [ ] tabs have solid color marker instead of gradient
- [ ] calculator buttons are too wide
- [ ] popup menus seem too rounded (but that's probably subjective, they look fine as is)
- [ ] panel applet grabber is too dark

**Comment by blue-devil-93** [comment]
@vkareh do a last try for fixing the  calculator buttons today or tommorow.
Then it can be merged with master branch i believe.

**Comment by blue-devil-93** [comment]
window>box.vertical>box.vertical>box.vertical>grid >button seems to be the
buttons. What is missing is the specific definition for Math Window.

**Comment by raveit65** [comment]
This is working for me, without the 2 grid between box.vertical and button.text-button
```
window > box.vertical > box.vertical > box.vertical button.text-button {
   background-color: black;
{
```
Tested with gtk3-inspector.

**Comment by blue-devil-93** [comment]
I did now fix all the issues mentioned.

**Comment by raveit65** [comment]
Is there a reason why you merge it without further testing by other team member?
You are new in the team don't do that!
I didn't gave you access to merge your own PR.

**Comment by raveit65** [comment]
Please make a new PR.

**Pull request merged**